backend/app/services/currency_converter.py

```
@@ -0,0 +1,141 @@
+        return Decimal("1.0")
+
+    try:
+        # Use PriceFetcher to get exchange rate
+        # Note: Yahoo Finance uses base/quote format, so we need to call it with from_currency as base
+        logger.info(f"Fetching exchange rate: {from_currency} -> {to_currency}")
+
+        # Run the async function in a sync context
+        rate = asyncio.run(_fetch_rate_async(from_currency, to_currency))
+
+        if rate is None:
+            raise ValueError(f"Failed to fetch exchange rate for {from_currency}/{to_currency}")
```

Ruff (0.14.0)

Abstract `raise` to an inner function

(TRY301)